dolphinscheduler-task-plugin/dolphinscheduler-task-datax/src/main/java/org/apache/dolphinscheduler/plugin/task/datax/DataxTask.java

```diff
@@ -419,6 +438,58 @@ private String buildShellCommandFile(String jobConfigFilePath, Map<String, Prope
         return fileName;
     }
 
+    private String buildYarnShellCommandFile(String jobConfigFilePath, Map<String, Property> paramsMap, String dataxHdfsPath, String dataxOnYarnDependJar)
```

Comment:
Since the execution of many DS task plugins is based on `ShellExecutor`, would u like to improve it so that those tasks could also run shell command in yarn containers? BTW, in this PR, it seems `datax-on-yarn-1.0.0.jar` is built from code in an individual repo and this might not be a good practice. https://github.com/apache/dolphinscheduler/blob/dev/dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/ShellCommandExecutor.java

Reply:
The improvement of `ShellCommandExecutor` I mentioned above is just a suggestion not a requirement here. You don't need to do it in this PR : )

Reply:
Because yarn submission needs to use a lot of hadoop and yarn dependencies, it is easy to cause dependencies